var/spack/repos/builtin/packages/julia/package.py

```diff
@@ -30,48 +32,42 @@ class Julia(Package):
     version('0.4.5', sha256='cbf361c23a77e7647040e8070371691083e92aa93c8a318afcc495ad1c3a71d9')
     version('0.4.3', sha256='2b9df25a8f58df8e43038ec30bae195dfb160abdf925f3fa193b59d40e4113c5')
 
-    # TODO: Split these out into jl-hdf5, jl-mpi packages etc.
-    variant("cxx", default=False, description="Prepare for Julia Cxx package")
-    variant("hdf5", default=False, description="Install Julia HDF5 package")
-    variant("mpi", default=True, description="Install Julia MPI package")
-    variant("plot", default=False,
-            description="Install Julia plotting packages")
-    variant("python", default=False,
-            description="Install Julia Python package")
-    variant("simd", default=False, description="Install Julia SIMD package")
-    variant("mkl", default=False, description="Use Intel MKL")
+    variant('cxx', default=False, description='Prepare for Julia Cxx package')
+    variant('mkl', default=False, description='Use Intel MKL')
 
     patch('gc.patch', when='@0.4:0.4.5')
     patch('openblas.patch', when='@0.4:0.4.5')
     patch('armgcc.patch', when='@1.0.0:1.1.1 %gcc@:5.9 target=aarch64:')
 
     variant('binutils', default=sys.platform != 'darwin',
-            description="Build via binutils")
+            description='Build via binutils')
 
     # Build-time dependencies:
-    # depends_on("awk")
-    depends_on("m4", type="build")
-    # depends_on("pkgconfig")
+    # depends_on('awk')
+    depends_on('m4', type='build')
+    # depends_on('pkgconfig')
+    # Python only needed to build LLVM?
+    depends_on('python@2.7:2.8', type='build', when='@:1.1')
+    depends_on('python@2.7:', type='build', when='@1.2:')
 
     # Combined build-time and run-time dependencies:
     # (Yes, these are run-time dependencies used by Julia's package manager.)
-    depends_on("binutils", when='+binutils')
-    depends_on("cmake @2.8:")
-    depends_on("curl")
-    depends_on("git", when='@:0.4')
-    depends_on("git", when='@release-0.4')
-    depends_on("openssl")
-    depends_on("python@2.7:2.8")
-    depends_on("mkl", when='+mkl')
+    depends_on('binutils', when='+binutils')
+    depends_on('cmake @2.8:')
```

Comment:
How many of these dependencies are only needed at build time?

Reply:
MKL is definitely needed as it is linked. I am not sure about curl and cmake, nor am I sure about the origin of the comment with regard to needing them for the julia package manager. Julia is on its third iteration of package manager so those may need to be version constrained. I would have to look into that more so I left those as is.

Reply:
It looks to me like curl could be a 'build, run' dependency for versions <1 and a 'build' dependency for versions >= 1. However, the build would not be able to find the curl headers if curl was not a link dependency.